specifyweb/stored_queries/execution.py

```diff
@@ -587,6 +587,7 @@ def build_query(session, collection, user, tableid, field_specs,
         collection=collection,
         objectformatter=ObjectFormatter(collection, user, replace_nulls),
         query=session.query(func.group_concat(id_field.distinct(), separator=',')) if distinct else session.query(id_field),
+        detect_cycles=settings.DEBUG
```

Comment:
`detect_cycles` is a bad variable name as it's supposed to always detect cycles, but then to only print debug information if this setting is true.

so really, this setting should be more like `debug=`, but probably shouldn't even exist because you can:
1. just reference settings.DEBUG directly in place where you were going to read this variable
2. OR remove this debug functionality entirely if you think it's not too likely that someone would benefit from this again in the future (and really, we should be using a python debugger or an SQL alchemy/MySQL debugger if there is such a thing)

Reply:
> just reference settings.DEBUG directly in place where you were going to read this variable

Hmm nice idea, the only problem is that unit tests do not necessarily run in debug mode. They, for example, don't run right now in debug mode. I have a unit test that detects cycles. I want to make sure it detects the correct cycle (the full path)

Reply:
Although, you are right about variable naming.

> OR remove this debug functionality entirely if you think it's not too likely that someone would benefit from this again in the future (and really, we should be using a python debugger or an SQL alchemy/MySQL debugger if there is such a thing)

Then that would mean only devs will be able to figure it out what's happening. Right now, on the test panel, it would show the cycle detected, so anyone on our team can figure out the faulty formatter definition.